code/controllers/subsystem/processing/quirks.dm

```diff
@@ -65,14 +64,14 @@ PROCESSING_SUBSYSTEM_DEF(quirks)
 		human.hardcore_survival_score = cli.prefs.hardcore_survival_score //Only do this if we actually asign quirks, to prevent sillicons etc from getting the points.
 
 	// Assign wayfinding pinpointer granting quirk if they're new
-	if(cli.get_exp_living(TRUE) < EXP_ASSIGN_WAYFINDER && !user.has_quirk(/datum/quirk/needswayfinder))
-		user.add_quirk(/datum/quirk/needswayfinder, TRUE)
+	if(cli.get_exp_living(TRUE) < EXP_ASSIGN_WAYFINDER && !user.has_quirk(/datum/quirk/item_quirk/needswayfinder))
```

Comment:
do you need this specific has_quirk check? add_quirk will already check if they have the trait or not, and if they do, then they won't be given the quirk.

Reply:
Probably not, but I don't feel it's harmful to explicitly prevent **attempting** to add the quirk when it already exists on the user and aids clarity in the intent of the code.

Reply:
No harm, and it's good not to rely too much on checks ahead in general.